### DISCUSSION TITLE
using fork with pcntl_signal/pcntl_alarm

### DISCUSSION THREAD

**Discussion by henzeb** [spatie/fork#31] [Ideas]
When using Fork together with pcntl calls like pcntl_signal (SIGTERM/SIGINT) or pcntl_alarm, I occasionally get the following error:

`socket_select(): Unable to select [4]: Interrupted system call`

This is a known issue and the common solution is to check for this error and retry again.

**Comment by freekmurze** [comment]
Feel free to PR improvements.
